### PR TITLE
NATS OpenTelemetry tracing

### PR DESCRIPTION
**Pull request by mtmk**
Updated NATS version with OTel support and enabled tracing in component.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2861)

### CODE REVIEW

File: tests/Aspire.NATS.Net.Tests/ConformanceTests.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ public class ConformanceTests : ConformanceTests<INatsConnection, NatsClientSett
     private const string ConnectionSting = "nats://apire-host:4222";
```
Comment (eerhardt) [inline tests/Aspire.NATS.Net.Tests/ConformanceTests.cs:14]: Side question, any chance you can update these tests to use https://www.nuget.org/packages/Testcontainers.Nats? (in a separate PR) That way the tests run in CI and we can ensure they pass successfully. See 

* https://github.com/dotnet/aspire/pull/2345
* https://github.com/dotnet/aspire/pull/2718
* https://github.com/dotnet/aspire/pull/2856

For examples.

File: src/Components/Aspire.NATS.Net/AspireNatsClientExtensions.cs [hunk 2]
```diff
@@ -19,6 +19,7 @@ namespace Microsoft.Extensions.Hosting;
 public static class AspireNatsClientExtensions
 {
     private const string DefaultConfigSectionName = "Aspire:NATS:Net";
+    private const string ActivityNameSource = "NATS.Net";
```
Comment (eerhardt) [inline src/Components/Aspire.NATS.Net/AspireNatsClientExtensions.cs:22]: Can you update 

* https://github.com/dotnet/aspire/blob/main/src/Components/Telemetry.md
* https://github.com/dotnet/aspire/blob/main/src/Components/Aspire_Components_Progress.md

With this information?

**Review by eerhardt — COMMENTED** [review]
Looks good. We just need to update our documentation.

**Comment by eerhardt** [comment]
> Unable to find package NATS.Net with version (>= 2.1.3)

I've kicked off getting this new version mirrored to our feeds.

File: tests/Aspire.NATS.Net.Tests/ConformanceTests.cs (hunk 1 above)
Comment (mtmk) [reply tests/Aspire.NATS.Net.Tests/ConformanceTests.cs:14]: that was fairly easy. didn't know about testcontainers. thanks for that
* #2886

**Review by eerhardt — APPROVED** [review]
Nice! LGTM. Thanks for the contribution.

**Pull request merged**